convert2rhel/unit_tests/toolopts_test.py

```diff
@@ -145,3 +146,145 @@ def test_both_disable_submgr_and_no_rhsm_options_work(argv, raise_exception, no_
         convert2rhel.toolopts.CLI()
 
     assert convert2rhel.toolopts.tool_opts.no_rhsm == no_rhsm_value
+
+
+@pytest.mark.parametrize(
+    ("argv", "content", "output"),
+    (
+        (
+            mock_cli_arguments([""]),
+            "[subscription_manager]\npassword=conf_pass",
+            {"password": "conf_pass", "activation_key": None},
+        ),
+        (
+            mock_cli_arguments(["-p", "password"]),
+            "[subscription_manager]\nactivation_key=conf_key",
+            {"password": "password", "activation_key": None},
+        ),
+        (
+            mock_cli_arguments(["-k", "activation_key"]),
+            "[subscription_manager]\nactivation_key=conf_key",
+            {"password": None, "activation_key": "activation_key"},
+        ),
+        (
+            mock_cli_arguments(["-k", "activation_key"]),
+            "[subscription_manager]\npassword=conf_pass",
+            {"password": "conf_pass", "activation_key": "activation_key"},
+        ),
+        (
+            mock_cli_arguments(["-k", "activation_key", "-p", "password"]),
+            "[subscription_manager]\npassword=conf_pass\nactivation_key=conf_key",
+            {"password": "password", "activation_key": "activation_key"},
+        ),
+        (
+            mock_cli_arguments([""]),
+            "[subscription_manager]\npassword=conf_pass\nactivation_key=conf_key",
+            {"password": "conf_pass", "activation_key": "conf_key"},
+        ),
+    ),
+)
+def test_config_file(argv, content, output, monkeypatch, tmp_path):
+    # After each test there were left data from previous
+    # Re-init needed delete the set data
+    convert2rhel.toolopts.tool_opts.__init__()
```

Comment:
No need to change this but I thought I should note that, in the future, I'm going to be introducing a fixture to conftest.py that partially handles this.  It looks like this:

``` python
@pytest.fixture
def global_tool_opts(monkeypatch):
    local_tool_opts = toolopts.ToolOpts()
    monkeypatch.setattr(toolopts, "tool_opts", local_tool_opts)
    return local_tool_opts
```

it's a little weird, though, because all of our modules use the global tool_opts with this sort of import: `from convert2rhel.toolopts import tool_opts` so we also have to monkeypatch `tool_opts` in the specific module.  I do that like this:

``` python
from convert2rhel import subscription

@pytest.fixture
def tool_opts(global_tool_opts, monkeypatch):
    monkeypatch.setattr(subscription, "tool_opts", global_tool_opts)
    return global_tool_opts

def test_subscribe_system(tool_opts):
    tool_opts.username = "user"
    tool_opts.password = "pass"
    assert subscription.subscribe_system()
```

(Like I said, no need to change this now.  Just something to be aware of when you port code to pytest in the future).